core/domain/classifier_domain_test.py

```diff
@@ -269,3 +269,59 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected training_data to be a list')):
             training_job.validate()
+
+
+class ClassifierExplorationMappingDomainTests(test_utils.GenericTestBase):
+    """Tests for the ClassifierExplorationMapping domain."""
+
+    def _get_mapping_from_dict(self, mapping_dict):
+        mapping = classifier_domain.ClassifierExplorationMapping(
+            mapping_dict['exp_id'],
+            mapping_dict['exp_version'],
+            mapping_dict['state_name'],
+            mapping_dict['classifier_id'])
+
+        return mapping
+
+    def test_to_dict(self):
+        expected_mapping_dict = {
+            'exp_id': 'exp_id1',
+            'exp_version': 2,
+            'state_name': 'state_name1',
+            'classifier_id': 'classifier_id1'
+        }
+        observed_mapping = self._get_mapping_from_dict(
+            expected_mapping_dict)
+        self.assertDictEqual(expected_mapping_dict,
+                             observed_mapping.to_dict())
+
+    def test_validation(self):
```

Comment:
@pranavsid98, I'm not sure why you say that we require utf-8 strings in the logic layer. AFAIK the storage layer IDs is the only place we require str, and in general we should be using unicode throughout. For example, try going to /admin and entering some unicode characters in one of the config property values. If you log the type in the backend controller you will see that it is unicode, not str.

In general, in a logic layer, I think it makes sense to handle native values, and not some encoding of them. You only encode when you need to transform it for some special purpose (e.g. GAE storage does not accept unicode IDs).